### ISSUE TITLE
 'Settings(settings)' accepts too few arguments to be used as a decorator here.

### ISSUE THREAD

**Issue by jwulf** [xuezier/grpc-server-ts#7]
```
 'Settings(settings)' accepts too few arguments to be used as a decorator here. Did youmean to call it first and write '@Settings(settings)()'?
```

<img width="511" alt="screen shot 2018-11-24 at 2 58 53 pm" src="https://user-images.githubusercontent.com/406975/48964725-89a53180-eff9-11e8-9ce0-e20be05db79f.png">

With version 1.1.0 of the package. I get this error when compiling.

**Comment by jwulf** [comment]
Fixed with #8

**Comment by xuezier** [comment]
@jwulf 

Okay, I found the mistake.
It's because the *Setting* function returns a decorator that does not require parameters.
I have improved it.

It's not a bug, but it's a nonstandard of decorator defining.

**Comment by xuezier** [comment]
> Fixed with #8

OK ， I just saw this mr.
thank you!
